src/ckeditor.jsx

```diff
@@ -51,60 +94,97 @@ export default class CKEditor extends React.Component {
 	}
 
 	_initializeEditor() {
-		this.props.editor
-			.create( this.domContainer.current, this._getConfig() )
-			.then( editor => {
-				this.editor = editor;
-
-				if ( 'disabled' in this.props ) {
-					editor.isReadOnly = this.props.disabled;
-				}
-
-				if ( this.props.onInit ) {
-					this.props.onInit( editor );
-				}
-
-				const modelDocument = editor.model.document;
-				const viewDocument = editor.editing.view.document;
-
-				modelDocument.on( 'change:data', event => {
-					/* istanbul ignore else */
-					if ( this.props.onChange ) {
-						this.props.onChange( event, editor );
+		const creator = ( el, config ) => {
+			return this.props.editor.create( el, config )
+				.then( editor => {
+					if ( 'disabled' in this.props ) {
+						editor.isReadOnly = this.props.disabled;
 					}
-				} );
 
-				viewDocument.on( 'focus', event => {
-					/* istanbul ignore else */
-					if ( this.props.onFocus ) {
-						this.props.onFocus( event, editor );
-					}
+					const modelDocument = editor.model.document;
+					const viewDocument = editor.editing.view.document;
+
+					modelDocument.on( 'change:data', event => {
+						/* istanbul ignore else */
+						if ( this.props.onChange ) {
+							this.props.onChange( event, editor );
+						}
+					} );
+
+					viewDocument.on( 'focus', event => {
+						/* istanbul ignore else */
+						if ( this.props.onFocus ) {
+							this.props.onFocus( event, editor );
+						}
+					} );
+
+					viewDocument.on( 'blur', event => {
+						/* istanbul ignore else */
+						if ( this.props.onBlur ) {
+							this.props.onBlur( event, editor );
+						}
+					} );
+
+					// The `onReady` callback should be fired once the `editor` property
+					// can be reached from the `<ckeditor>` component.
+					// Ideally this part should be moved to the watchdog item creator listeners.
+					setTimeout( () => {
+						if ( this.props.onReady ) {
+							this.props.onReady( { editor: this.editor } );
+						}
+					} );
+
+					return editor;
 				} );
+		};
 
-				viewDocument.on( 'blur', event => {
-					/* istanbul ignore else */
-					if ( this.props.onBlur ) {
-						this.props.onBlur( event, editor );
-					}
-				} );
-			} )
-			.catch( error => {
-				const onErrorCallback = this.props.onError || console.error;
+		const onError = ( { error, phase = 'runtime', causesRestart = false } ) => {
+			const onErrorCallback = this.props.onError || console.error;
 
-				onErrorCallback( error );
+			onErrorCallback( {
+				error,
+				phase,
+				willEditorRestart: causesRestart
 			} );
```

Comment:
```suggestion
			onErrorCallback( error, {
				phase,
				willEditorRestart: causesRestart
			} );
```

WDYT? If we won't modify the API of `onReady` event, users after upgrading the package will have less to fix. But it's just thought, we can leave it.

Reply:
:+1: